File: C/assign_cookies.c
void sort(int* arr, int size) {
    for(int i = 0; i < size-1; i++) {
        for(int j = i+1; j < size; j++) {
            if(arr[i] > arr[j]) {
                int tmp = arr[i];
                arr[i] = arr[j];
                arr[j] = tmp;
            }
        }
    }
}

int findContentChildren(int* g, int gSize, int* s, int sSize){
    if(gSize == 0 || sSize == 0) return 0;

    sort(g, gSize);
	sort(s, sSize);
    int result = 0;
    int child_id = 0;
        
    for(int i = 0; i < sSize; i++) {
        if(child_id >= gSize) return result;
        if(s[i] >= g[child_id]) {
            result ++;
            child_id ++;
        }
    }
        
    return result;
}

File: Python3/assign_cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        if not g or not s: return 0
        
        g.sort()
        s.sort()
        result = 0
        child_id = 0
        
        for cookies in s:
            if child_id >= len(g):
                return result
            if cookies >= g[child_id]:
                result += 1
                child_id += 1
        
        return result
